ci_tools/check_slots.py

```diff
@@ -26,82 +28,132 @@
 missing_attribute_nodes = []
 missing_from_all = []
 
+ErrorCollection = {
+    'missing_all':[],
+    'non_alphabetical_all':[],
+    'missing_slots':[],
+    'overridden_slots':[],
+    'missing_attribute_nodes':[],
+    'missing_from_all':[]
+}
+
+def extract_dict_elements(input_dict):
+    """
+    Extracts specific elements from the input dictionary based on the given keys and returns a new dictionary.
+
+    Args:
+        input_dict (dict): The input dictionary from which elements will be extracted.
+
+    Returns:
+        dict: A new dictionary containing the extracted elements.
+    """
+    output_dict = {'title':"Pyccel_lint", 'summary':"Check Slots:\n\n",'annotations':[]}
+    for values in input_dict.items():
+        first_iteration = True
+        for value in values[-1]:
+            if value:
+                if first_iteration:
+                    output_dict['summary'] += f"{value['title']} :\n\n"
+                    first_iteration = False
+                output_dict['summary'] += f"\t-{value['message']}\n\n"
+                output_dict['annotations'].append(value['annotations'])
```

Comment:
The `first_iteration` check seems unnecessary to me. If blocks in for loops are generally a bad idea as they break pipelines which can slow down code significantly. It is good practice to avoid them where possible. Can this be reduced to:
```suggestion
        value_list = values[-1]
        if value_list:
            output_dict['summary'] += f"{value_list[0]['title']} :\n\n"
            output_dict['summary'] += "\n\n".join(f"-  {value['message']}" for value in value_list)
            output_dict['summary'] += "\n\n"
            output_dict['annotations'].extend(value['annotations'] for value in value_list)
```